### PR TITLE
[JENKINS-58578] Force the setting of the private ip also for public subnet

### PR DESCRIPTION
**Pull request by thoulen**
JENKINS-58578 Force the setting of the private ip also for public subnet

### CODE REVIEW

**Comment by andgoldin** [comment]
This should probably also be updated in the `provisionSpot` function as well: [Lines 950-954](https://github.com/jenkinsci/ec2-plugin/blob/f327c01986aa99033e4e514ceb01e656c485eb3c/src/main/java/hudson/plugins/ec2/SlaveTemplate.java#L950-L954)

**Comment by jvz** [comment]
Is this unit testable?

File: src/main/java/hudson/plugins/ec2/SlaveTemplate.java [hunk 1]
```diff
@@ -633,11 +633,12 @@ private void logProvisionInfo(String message) {
             }
         }
 
+        net.setAssociatePublicIpAddress(false);
```
Comment (res0nance) [inline src/main/java/hudson/plugins/ec2/SlaveTemplate.java]:
```suggestion
        net.setAssociatePublicIpAddress(getAssociatePublicIp());
```
Will allow you to remove the if

File: src/main/java/hudson/plugins/ec2/SlaveTemplate.java [hunk 2]
```diff
@@ -946,11 +947,12 @@ private void setupCustomDeviceMapping(List<BlockDeviceMapping> deviceMappings) {
             launchSpecification.setKeyName(keyPair.getKeyName());
             launchSpecification.setInstanceType(type.toString());
 
+            net.setAssociatePublicIpAddress(false);
```
Comment (res0nance) [inline src/main/java/hudson/plugins/ec2/SlaveTemplate.java]:
```suggestion
        net.setAssociatePublicIpAddress(getAssociatePublicIp());
```
Same here

**Comment by varyvol** [comment]
Do you think this could be merged?

**Pull request merged**